File: src/main/java/com/github/unchama/seichiassist/commands/MineHeadCommand.kt
package com.github.unchama.seichiassist.commands

import com.github.unchama.contextualexecutor.asNonBlockingTabExecutor
import com.github.unchama.targetedeffect.asMessageEffect
import com.github.unchama.seichiassist.commands.contextual.builder.BuilderTemplates.playerCommandBuilder
import com.github.unchama.seichiassist.util.StaticGachaPrizeFactory
import com.github.unchama.seichiassist.util.Util
import org.bukkit.ChatColor

object MineHeadCommand {
  val executor = playerCommandBuilder
      .execution { context ->
        Util.addItemToPlayerSafely(context.sender, StaticGachaPrizeFactory.getMineHeadItem())

        "${ChatColor.GREEN}専用アイテムを付与しました。".asMessageEffect()
      }
      .build()
      .asNonBlockingTabExecutor()
}

File: src/main/scala/com/github/unchama/seichiassist/commands/MineHeadCommand.scala
package com.github.unchama.seichiassist.commands

import cats.effect.IO
import com.github.unchama.minecraft.actions.OnMinecraftServerThread
import com.github.unchama.seichiassist.commands.contextual.builder.BuilderTemplates.playerCommandBuilder
import com.github.unchama.seichiassist.subsystems.gachaprize.bukkit.factories.BukkitStaticGachaPrizeFactory
import com.github.unchama.seichiassist.util.InventoryOperations
import com.github.unchama.targetedeffect.commandsender.MessageEffect
import org.bukkit.ChatColor._
import org.bukkit.command.TabExecutor
import org.bukkit.entity.Player

class MineHeadCommand(implicit ioOnMainThread: OnMinecraftServerThread[IO]) {

  import com.github.unchama.targetedeffect._

  val effect: TargetedEffect[Player] =
    SequentialEffect(
      InventoryOperations.grantItemStacksEffect(BukkitStaticGachaPrizeFactory.mineHeadItem),
      MessageEffect(s"${GREEN}専用アイテムを付与しました。")
    )

  val executor: TabExecutor =
    playerCommandBuilder[Nothing].execution { _ => IO.pure(effect) }.build().asNonBlockingTabExecutor()
}
